models/bin.js

```diff
@@ -2,3 +2,41 @@
 
 // Create a bin using postgres
 // Fetch all bins using postgres
+
+const client = require('../db/pg.js');
+
+async function allBins() {
+  try {
+    const bins = await client.query("SELECT * FROM bins");
+    return bins.rows;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+
+async function binByUUID(id) {
+ try{
+   const bin = await client.query("SELECT id FROM bins WHERE uuid = $1", [id]); 
+   return bin.rows[0];
+ } catch (error) {
+   console.error(error);
+ } 
+}
+
+async function newBin() {
+  try {
+    await client.query("INSERT INTO bins DEFAULT VALUES");
+    const newUUID = await client.query("SELECT uuid FROM bins ORDER BY id DESC LIMIT 1");
```

Comment:
There might be some risk in doing this in two queries (especially if there are many people creating bins simultaneously). Looks like you can do something like the following: 

```
INSERT INTO users (name, email)
VALUES ('John Doe', 'johndoe@example.com')
RETURNING id, name;
```

What do you think?

Reply:
Yes this syntax works great, I used in the crud app for the individual VPS project. You can tack `RETURNING` clauses onto `UPDATE` statements too!